### PR TITLE
KN-41270 Added handling of server errors during API calls.

### PR DESCRIPTION
**Pull request by unicrus**

### CODE REVIEW

File: Knoema.Client/Client.cs [hunk 1]
```diff
@@ -117,10 +120,27 @@ private HttpClient GetApiClient(int timeout = HttpClientTimeout)
...
 
+        private static void EnsureSuccessApiCall(HttpResponseMessage postResponse)
+        {
+            if (!postResponse.IsSuccessStatusCode)
+            {
+                var error = "";
+                if (postResponse.Content != null)
+                {
+                    error = postResponse.Content.ReadAsStringAsync().Result;
+                    error = Regex.Replace(error, "<style>(.|\n)+?</style>|<[^>]+>", String.Empty, RegexOptions.Multiline);
```
Comment (AlexPhilippov) [inline Knoema.Client/Client.cs]: Will it work for 404 page or 403?

Comment (unicrus) [reply Knoema.Client/Client.cs]: It works. Displays the error

> Remote server returned error 404

File: Knoema.Client/Client.cs [hunk 2]
```diff
@@ -117,10 +120,27 @@ private HttpClient GetApiClient(int timeout = HttpClientTimeout)
...
+                    error = Regex.Replace(error, @"\r\n\s*\r\n", "\r\n").Trim();
+                }
+                throw new Exception(
+                    String.Format("Remote server returned error {0}\r\n\r\n{1}", (int)postResponse.StatusCode, error));
```
Comment (PavelPimenov) [inline Knoema.Client/Client.cs]: environment.newline instead of \r\n

Comment (unicrus) [reply Knoema.Client/Client.cs]: It does not depend on environment where Knoema.Client is used. It's server-generated error message.

File: Knoema.Client/Client.cs [hunk 3]
```diff
@@ -117,10 +120,27 @@ private HttpClient GetApiClient(int timeout = HttpClientTimeout)
...
+            if (!postResponse.IsSuccessStatusCode)
+            {
+                var error = "";
+                if (postResponse.Content != null)
+                {
+                    error = postResponse.Content.ReadAsStringAsync().Result;
+                    error = Regex.Replace(error, "<style>(.|\n)+?</style>|<[^>]+>", String.Empty, RegexOptions.Multiline);
+                    error = Regex.Replace(error, @"\r\n\s*\r\n", "\r\n").Trim();
+                }
+                throw new Exception(
```
Comment (FluttercordKN) [inline Knoema.Client/Client.cs]: Do not throw System.Exception

File: Knoema.Client/Client.cs (hunk 2 above)
Comment (unicrus) [reply Knoema.Client/Client.cs]: Changed.

File: Knoema.Client/Client.cs (hunk 3 above)
Comment (unicrus) [reply Knoema.Client/Client.cs]: Switched to HttpException.

**Pull request merged**